app/scripts/store/dimStoreItem.directive.js

```diff
@@ -67,11 +68,26 @@
           ngDialog.closeAll();
 
           if (!dimLoadoutService.dialogOpen) {
-            var bottom = ($(element).offset().top < 300) ? ' move-popup-bottom' : '';
-            var right = ((($('body').width() - $(element).offset().left - 320) < 0) ? ' move-popup-right' : '');
+
+            var settings = dimSettingsService.getSettings().$$state.value,
+              verbosePerks = settings.verbosePerks,
+              minOffset = 300 + (item.perks.length * 44 * verbosePerks),
+              perkStateString = verbosePerks ? 'verbose-perks' :'quiet-perks',
+
+              bottom = ($(element).offset().top < minOffset) ? ' move-popup-bottom' : '',
+              right = ((($('body').width() - $(element).offset().left - 320) < 0) ? ' move-popup-right' : ''),
+              templateString =
+                `
```

Comment:
Appreciate the work but I would like the code to be consistent since we're maintaining it.  We'll are on ES6 in v4 and we'll take these patterns with us there.